SRCH-504 - Remove all code and tests related to Instagram

danswick opened the pull request:
## Summary
This PR removes dead code and tests related to search-gov's now-deprecated Instagram functionality. Follow-up work related to db migrations will be done in #966. 
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional.

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures: rubocop flagged an issue in an unrelated part of an edited file, so I've manually marked the issue as "Approved."

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock.
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
 
- [x] If your target branch is NOT `main`, specify the reason:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- ⚠️ You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop). **Now that "squash-and-commit" is enforced by default, this step will happen when this PR is merged**.

- ⚠️ Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.  **Now that "squash-and-commit" is enforced by default, this step will happen when this PR is merged**.
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

danswick commented on lib/middlewares/mrashify.rb:
```diff
@@ -7,8 +7,10 @@
+# whose current version conflicted with the dependencies in a now-removed instagram gem...
+# which was deprecated in 2015.
+# Fully removing this custom middleware will require some additional work
+# and has been put off until a later date.
```
Per discussion with @MothOnMars, we may want to remove `mrashify` at some point, but it's not a risk to leave it in and is out of scope for this PR. For now, I've just updated the note at the top of the file for future reference.

lorawoodford replied:
Sounds good.  I see the https://cm-jira.usa.gov/browse/SRCH-3228 follow on.

danswick commented on spec/engines/oasis_search_spec.rb:
```diff
@@ -15,7 +15,7 @@
       normalized_response = image_search.execute_query
       expect(normalized_response.start_record).to eq(1)
-      expect(normalized_response.end_record).to eq(10)
+      expect(normalized_response.end_record).to eq(9)
```
The fixture had one entry related to instagram, which has been removed. Noting the change here in case others think it's problematic for the expected value to be different from the [`size` parameter above](https://github.com/GSA/search-gov/blob/1a8f46e548c1b0020dcdab5c1ca7c26ae1852a08/spec/engines/oasis_search_spec.rb#L10).

danswick commented on spec/fixtures/html/home_page_with_social_media_urls.html:
```diff
@@ -1537,7 +1537,6 @@ <h3>Stay Connected</h3>
 <li class="flickr"><a href="http://flickr.com/photos/whitehouse">Flickr</a></li>
 <li class="youtube"><a href="http://www.youtube.com/watch?v=1JqR3GVqib4&list=UUDsO-0Yo5zpJk575nKXgMVA&index=1&feature=plcp">YouTube</a></li>
 <li class="youtube"><a href="http://www.youtube.com/whitehouse1?watch=0">YouTube</a></li>
-<li class="instagram"><a href="http://www.instagram.com/whitehouse">Instagram</a></li>
```
This removal isn't really necessary, now that I'm looking at the diff.

danswick commented on spec/requests/image_searches_spec.rb:
```diff
@@ -1,7 +1,7 @@
 
 describe '/search/images' do
-  fixtures :affiliates, :instagram_profiles
+  fixtures :affiliates, :flickr_profiles
```
I decided to swap flickr in here to preserve the images use case. Open to other ideas of this isn't appropriate.

lorawoodford replied:
👍

lorawoodford reviewed (changes requested):
@danswick The current code changes look fine, but a few updates:

1. Please add a brief description to the Summary section of the PR comment;
2. Implement @MothOnMars 's change request to call out the JIRA ticket for the mrashify comment; and, 
3. Choose how you would like to resolve the other outstanding comment re: `spec/fixtures/html/home_page_with_social_media_urls.html` (I have no preference).

danswick commented:
Thank you for the review, @lorawoodford! I addressed points 1 and 2 and decided to revert my changes from point 3. Would love another review when you get a chance 🙏

lorawoodford reviewed (approved):
LGTM
